src/lock.rs

```diff
@@ -0,0 +1,86 @@
+use std::fs::write;
+use std::path::{Path, PathBuf};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{files, util::Fallible};
+use std::collections::{HashMap, HashSet};
+
+pub fn code_changed<P: AsRef<Path>>(lock_file: P) -> Fallible<bool> {
+    if lock_file.as_ref().exists() && lock_file.as_ref().is_file() {
```

Comment:
The second check includes the first. More generally, it should suffice to just call `Lock::read` and handle any IO and parsing errors there.

Reply:
My intention here was to skip the test if no lock file exists, not to prevent errors. The first condition can be removed, but reading could also fail due to a malformed toml file.

Reply:
> The first condition can be removed, but reading could also fail due to a malformed toml file.

But do you want to handle this differently than a non-existant file?

Reply:
I think a mal-formed file should result in an error. A missing files could mean that 1) this is the first run with reverse-mode enabled, or 2) the user deleted the file to get rid of a lock. I these cases, it shoud just skip silently and create the lock file after processing.

Reply:
Valid, but then I would suggest to include a suggestion to delete the malformed file (or use the `--force` flag?) in the error message so that users know how to handle a malformed file.